server/src/lib.rs

```diff
@@ -243,19 +288,30 @@ fn log_response(addr: &SocketAddr, method: &str, uri: &str, start: &Instant, sta
     );
 }
 
+
 async fn handle(
     server: ServerSettings,
     collector: Collector,
     db: Db,
     subscriptions: Subscriptions,
     heartbeat_tx: mpsc::Sender<WriteHeartbeatMessage>,
-    auth_mech: AuthenticationMechanism,
-    conn_state: Arc<Mutex<kerberos::State>>,
+    auth_infos: Option<AuthenticationInfos>,
     addr: SocketAddr,
     req: Request<Body>,
 ) -> Result<Response<Body>, Infallible> {
     let start = Instant::now();
 
+    let auth_infos = match auth_infos {
```

Comment:
This looks strange because you expect that if `auth_infos` is `None` then there is a `Tls` error. Also, I think this should be handled in `authenticate`.